File: backend/src/main/java/edu/cit/aligato/fortpointproperties/auth/dto/UpdateProfileRequest.java
package edu.cit.aligato.fortpointproperties.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class UpdateProfileRequest {
    @NotBlank(message = "Email is required")
    @Email(message = "Email should be valid")
    private String email;

    @NotBlank(message = "First name is required")
    private String firstname;

    @NotBlank(message = "Last name is required")
    private String lastname;

    @Size(max = 30, message = "Phone number must be 30 characters or fewer")
    private String phoneNumber;

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }
}

File: mobile/app/src/main/java/com/example/fortpointproperties/features/auth/data/UpdateProfileRequest.kt
package com.example.fortpointproperties.features.auth.data

import com.google.gson.annotations.SerializedName

data class UpdateProfileRequest(
    @SerializedName("email")
    val email: String,

    @SerializedName("firstname")
    val firstname: String,

    @SerializedName("lastname")
    val lastname: String,

    @SerializedName("phoneNumber")
    val phoneNumber: String? = null
)
